src/nnet3/nnet-test-utils.cc

```diff
@@ -793,7 +793,7 @@ void ComputeExampleComputationRequestSimple(
 
 static void GenerateRandomComponentConfig(std::string *component_type,
                                           std::string *config) {
-  int32 n = RandInt(0, 20);
+  int32 n = RandInt(0, 21);
```

Comment:
Yes. Except abs-min which I assumed unnecessary based on your comment (and also there was no such function already implemented).
Also I did not remove CollapseWithNext() and Resize() from ScaleComponent because I was not sure if you wanted them to be deleted too.